### PR TITLE
[conformance][API] Add QN implementation check to API conformnace

### PR DESCRIPTION
**Pull request by iefode**
### Details:
 - QN for ops based on "OpImplCheck". It contains simple generated graphs for all ops

### Tickets:
 - *66416*

### CODE REVIEW

**Comment by iefode** [comment]
@ilya-lavrenov Please take a look

File: src/tests/functional/plugin/shared/src/single_layer_tests/op_impl_check/op_impl_check.cpp [hunk 1]
```diff
@@ -47,6 +47,53 @@ void OpImplCheckTest::run() {
     }
 }
 
+void OpImplCheckTest::query_model() {
+    if (function == nullptr) {
+        GTEST_FAIL() << "Target function is empty!";
```
Comment (ilyachur) [inline src/tests/functional/plugin/shared/src/single_layer_tests/op_impl_check/op_impl_check.cpp:52]:
```suggestion
        GTEST_FAIL() << "Target model is empty!";
```

File: src/tests/functional/plugin/shared/src/single_layer_tests/op_impl_check/op_impl_check.cpp [hunk 2]
```diff
@@ -47,6 +47,53 @@ void OpImplCheckTest::run() {
     }
 }
 
+void OpImplCheckTest::query_model() {
```
Comment (ilyachur) [inline src/tests/functional/plugin/shared/src/single_layer_tests/op_impl_check/op_impl_check.cpp:50]: Why do we need to introduce a new method which is used only in one test? Can we move it to the test body?

File: src/tests/functional/shared_test_classes/src/base/ov_subgraph.cpp [hunk 3]
```diff
@@ -144,7 +144,9 @@ void SubgraphBaseTest::query_model() {
     for (auto&& res : queryNetworkResult) {
         actual.insert(res.first);
     }
-    ASSERT_EQ(expected, actual);
```
Comment (praasz) [inline src/tests/functional/shared_test_classes/src/base/ov_subgraph.cpp:147]: Why remove test assert?

Comment (iefode) [reply src/tests/functional/shared_test_classes/src/base/ov_subgraph.cpp:147]: Assert is not applicable for report. We should to catch exceptions.

File: src/tests/functional/plugin/shared/include/single_layer_tests/op_impl_check/op_impl_check_compile_model.hpp [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+namespace test {
+namespace subgraph {
+
+TEST_P(OpImplCheckTest, checkPluginImplementation) {
+    if (function == nullptr) {
+        GTEST_FAIL() << "Target model is empty!";
+    }
+
+    // in case of crash jump will be made and work will be continued
+    auto crashHandler = std::unique_ptr<CommonTestUtils::CrashHandler>(new CommonTestUtils::CrashHandler());
```
Comment (ilyachur) [inline src/tests/functional/plugin/shared/include/single_layer_tests/op_impl_check/op_impl_check_compile_model.hpp:17]:
```suggestion
    CommonTestUtils::CrashHandler crashHandler;
```

File: src/tests/functional/plugin/shared/include/single_layer_tests/op_impl_check/op_impl_check_compile_model.hpp [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+    // place to jump in case of a crash
+    int jmpRes = 0;
+#ifdef _WIN32
+    jmpRes = setjmp(CommonTestUtils::env);
+#else
+    jmpRes = sigsetjmp(CommonTestUtils::env, 1);
+#endif
+    if (jmpRes == CommonTestUtils::JMP_STATUS::ok) {
+        crashHandler->StartTimer();
```
Comment (ilyachur) [inline src/tests/functional/plugin/shared/include/single_layer_tests/op_impl_check/op_impl_check_compile_model.hpp:27]:
```suggestion
        crashHandler.StartTimer();
```

File: src/tests/functional/plugin/shared/include/single_layer_tests/op_impl_check/op_impl_check_query_model.hpp [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+namespace ov {
+namespace test {
+namespace subgraph {
+
+TEST_P(OpImplCheckTest, checkPluginImplementation) {
+        if (function == nullptr) {
+        GTEST_FAIL() << "Target model is empty!";
+    }
+
```
Comment (ilyachur) [inline src/tests/functional/plugin/shared/include/single_layer_tests/op_impl_check/op_impl_check_query_model.hpp:15]: indentation

File: src/tests/functional/plugin/shared/include/single_layer_tests/op_impl_check/op_impl_check_query_model.hpp [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+    // in case of crash jump will be made and work will be continued
+    auto crashHandler = std::unique_ptr<CommonTestUtils::CrashHandler>(new CommonTestUtils::CrashHandler());
```
Comment (ilyachur) [inline src/tests/functional/plugin/shared/include/single_layer_tests/op_impl_check/op_impl_check_query_model.hpp:17]:
```suggestion
    CommonTestUtils::CrashHandler crashHandler;
```

File: src/tests/functional/plugin/shared/include/single_layer_tests/op_impl_check/op_impl_check_query_model.hpp [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+                IE_THROW() << "Expected and actual results are different";
+            }
+            summary.updateOPsImplStatus(function, true);
+        } catch (...) {
+            summary.updateOPsImplStatus(function, false);
+            GTEST_FAIL() << "Error in the Core::query_model() method call!";
+        }
+    } else if (jmpRes == CommonTestUtils::JMP_STATUS::anyError) {
+        summary.updateOPsImplStatus(function, false);
+        IE_THROW() << "Crash happens";
```
Comment (ilyachur) [inline src/tests/functional/plugin/shared/include/single_layer_tests/op_impl_check/op_impl_check_query_model.hpp:51]:
```suggestion
        GTEST_FAIL() << "Crash happens";
```

File: src/tests/functional/plugin/shared/include/single_layer_tests/op_impl_check/op_impl_check_compile_model.hpp (hunk 4 above)
Comment (iefode) [reply src/tests/functional/plugin/shared/include/single_layer_tests/op_impl_check/op_impl_check_compile_model.hpp]: Fixed

File: src/tests/functional/plugin/shared/include/single_layer_tests/op_impl_check/op_impl_check_compile_model.hpp (hunk 5 above)
Comment (iefode) [reply src/tests/functional/plugin/shared/include/single_layer_tests/op_impl_check/op_impl_check_compile_model.hpp]: Fixed

File: src/tests/functional/plugin/shared/include/single_layer_tests/op_impl_check/op_impl_check_query_model.hpp (hunk 6 above)
Comment (iefode) [reply src/tests/functional/plugin/shared/include/single_layer_tests/op_impl_check/op_impl_check_query_model.hpp]: Fixed

File: src/tests/functional/plugin/shared/include/single_layer_tests/op_impl_check/op_impl_check_query_model.hpp (hunk 7 above)
Comment (iefode) [reply src/tests/functional/plugin/shared/include/single_layer_tests/op_impl_check/op_impl_check_query_model.hpp]: Fixed

File: src/tests/functional/plugin/shared/include/single_layer_tests/op_impl_check/op_impl_check_query_model.hpp (hunk 8 above)
Comment (iefode) [reply src/tests/functional/plugin/shared/include/single_layer_tests/op_impl_check/op_impl_check_query_model.hpp]: Fixed

**Review by ilyachur — APPROVED** [review]

**Review by praasz — APPROVED** [review]

**Pull request merged**